apps/api/src/queries/buildViewModelForResource/viewModels/base-resource.view-model.ts

```diff
@@ -32,4 +39,11 @@ export class BaseResourceViewModel extends BaseViewModel {
 
         this.contributions = [...new Set(contributionsWithDuplicates)];
     }
+    getAvailableCommands(): string[] {
+        throw new Error(`Not Implemented: Did you mean to project off the event-sourced view?`);
+    }
+
+    getCompositeIdentifier(): { type: AggregateType; id: AggregateId } {
```

Comment:
I wonder if this is really what we want? If we are projecting off domain models to build views, we could move the logic from the corresponding domain model to the view model. That said, all resource views will be event-sourced soon enough, so I agree with sidestepping the issue for now. But keep in mind that we will circle back to the non-categorizable (non-resource and non-note) aggregates, which may be state based \ project views off the domain.

Reply:
I added a comment about this.